### PR TITLE
#1019 feat: Display warnings when changed training subsets or not valid subsets sizes

### LINKED ISSUE #1019: UX Update on training flow revamp - subset distribution
The issue is that the subset proportions (train + validation + test) are determined at the task level, which means that once any model is trained, the subset distribution cannot be changed for any future model unless you train from scratch and enable "Reshuffle subsets."
 
In practice, this means that changing the subset distribution

Image

won't have any effect unless you also select "train from scratch" and "reshuffle subsets."
 Image
 
I believe that if we change the backend logic to fit the current UX, we might end up with inconsistencies in the training data, such as mixing the training set with the test set in subsequent rounds of training.
So, I think it's better to adjust the UX so that when the user changes the distribution, "train from scratch" and "reshuffle subsets" are also enabled, and a warning is prompted to the user

Context: https://github.com/open-edge-platform/geti/issues/955

### PR DESCRIPTION
**Pull request by dwesolow**
## 📝 Description

image

image

Closes #1019 

## ✨ Type of Change

Select the type of change your PR introduces:

- [X] 🚀 **New feature** – Non-breaking change which adds functionality
- [X] 🧪 **Tests**

## 🧪 Testing Scenarios

Describe how the changes were tested and how reviewers can test them too:

- [X] ✅ Tested manually
- [ ] 🤖 Run automated end-to-end tests

## ✅ Checklist

Before submitting the PR, ensure the following:

- [X] 🔍 PR title is clear and meaningful
- [ ] ✍️ PR description clearly explains the changes and their reason
- [ ] 📝 I have linked the PR to the corresponding GitHub Issues, if any
- [ ] 💬 I have commented my code, especially in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [X] ✅ I have added tests that prove my fix is effective or my feature works

### CODE REVIEW

**Review by pplaskie — DISMISSED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/use-train-model-state.hook.ts [hunk 1]
```diff
@@ -120,6 +122,11 @@ export const useTrainModelState = () => {
     });
 
     const isBasicMode = mode === TrainModelMode.BASIC;
+    const hasModels =
```
Comment (jpggvilaca) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/use-train-model-state.hook.ts]: np: hasSupportedModels , to make it very clear that we MIGHT have a model, but if it's SAM, we dont want it : P

Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/use-train-model-state.hook.ts]: Ok, updated

**Review by jpggvilaca — APPROVED** [review]

**Review by pplaskie — APPROVED** [review]
